server/src/routes/api/v2/wallets.js

```diff
@@ -222,4 +223,31 @@ router.post('/backup', auth.required, async (req, res, next) => {
   return res.json({ job: job.attrs })
 })
 
+/**
+ * @api {post} api/v2/wallets/foreign Create wallet contract for user on Ethereum
+ * @apiName CreateWalletForeign
+ * @apiGroup Wallet
+ * @apiDescription Creates wallet contract for the user on Ethereum
+ *
+ * @apiHeader {String} Authorization JWT Authorization in a format "Bearer {jwtToken}"
+ *
+ * @apiSuccess {Object} Started job data
+ */
+router.post('/foreign', auth.required, async (req, res, next) => {
+  const { phoneNumber, accountAddress } = req.user
+  const network = config.get('network.foreign.name')
+
+  const userWallet = await UserWallet.findOne({ phoneNumber, accountAddress })
+  if (!userWallet) {
+    const msg = `User ${phoneNumber}, ${accountAddress} doesn't have a wallet account on fuse yet, cannot create on ${network}`
+    return res.status(400).json({ error: msg })
+  }
+  if (userWallet.networks.includes[network]) {
+    const msg = `User ${phoneNumber}, ${accountAddress} already has wallet account: ${userWallet.walletAddress} on ${network}`
+    return res.status(400).json({ error: msg })
+  }
+  const job = await agenda.now('createForeignWallet', { userWallet })
```

Comment:
Please add `correlationId` to the agenda job (optional body param)
see other routes (mainly admin ones)